### ISSUE TITLE
Support custom database

### ISSUE THREAD

**Issue by excaptor** [IllDepence/anki_add_pitch_plugin#9]
The core of the request is obvious from the title - allow to supply user's own csv.

The idea is - everyone edits the same words, not covered by wadoku, in their decks. Instead, these edits can be done as custom user csv, with tech savvy users contributing their additions to this repo. That way, wadoku omissions can be filled in by users over time.

Even if you don't want to maintain this database, someone else may want, it's just a few lines in a code which would allow custom file to be used in addition to wadoku. (And probably a checkmark in preferences to enable/disable it to lower data contamination concern.)

I can look into making pull request if we agree on approach.

**Comment by IllDepence** [comment]
Thanks for the suggestion!

I like the idea of an optional, additional crowd sourced pattern DB. Here are some initial thoughts:

* As an alternative to a use/don't use checkbox/setting an easy solution would be to just define a file name, that, when existent, is also read (`user_pitchdb.csv` or sth like that)
    * That way people could just create/use their own custom DBs
    * This would also allow multiple groups to maintain their own shared custom DBs
* If making it a checkbox thing
    * Shipping the crowd sourced custom DB with the add-on via the Add-ons for Anki site would require frequent updates—kind of cumbersome
    * Alternatively the add-on could be set to download the most recent version of the crowd sourced custom DB

I think I prefer the hands-off way of just defining a file name and letting users go wild. Any thoughts?

**Comment by excaptor** [comment]
tl;dr I agree 👍 

As plugin maintainer you're probably right you don't want to have user-sourced database as part of the plugin.

As a user, I want see some complete solution where one half is support for the user-supplied file. Another is some way to maintain that file in a way where Anki users (and specifically plugin users) can discover all of the following: there's such a function, how to access the file and how to contribute to it.
I see having the file with the plugin as most obvious solution. Most likely - because I'm not familiar with how collaboration around Anki works. And yes, that approach would be a bit misplaced and there probably is a better way.

Anyway, wherever file will end up, it will need support in a plugin, so I'm totally for that half.

**Comment by IllDepence** [comment]
Addded

* a menu item to display the file location
* code to load it
* a first example entry that is not present in the wadoku data but in the ＮＨＫ 日本語発音アクセント辞典

Could you use the updated files from the repository (see commit above) and test it on your end?

**Comment by IllDepence** [comment]
Updated the version on ankiweb now.
